From: Simon McVittie <smcv@debian.org>
Date: Sun, 12 Jun 2016 10:25:21 -0400
Subject: Link libreaddir-rand to libdl

It uses dlsym(). There's no point in being extra-portable here
because OSTree only targets Linux anyway.

Signed-off-by: Simon McVittie <smcv@debian.org>

Closes: #336
Approved by: cgwalters
Applied-upstream: 2016.6, commit:b9e18b83fd195c980c5dffab4bedf3fa79b19a2e
---
 Makefile-tests.am | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Makefile-tests.am b/Makefile-tests.am
index b3d7514..c7e78cf 100644
--- a/Makefile-tests.am
+++ b/Makefile-tests.am
@@ -130,7 +130,10 @@ endif
 test_ltlibraries = libreaddir-rand.la
 libreaddir_rand_la_SOURCES = tests/readdir-rand.c
 libreaddir_rand_la_CFLAGS = $(OT_INTERNAL_GIO_UNIX_CFLAGS)
-libreaddir_rand_la_LIBADD = $(OT_INTERNAL_GIO_UNIX_LIBS)
+libreaddir_rand_la_LIBADD = \
+	-ldl \
+	$(OT_INTERNAL_GIO_UNIX_LIBS) \
+	$(NULL)
 libreaddir_rand_la_LDFLAGS = -avoid-version
 if !ENABLE_INSTALLED_TESTS
 libreaddir_rand_la_LDFLAGS += -rpath $(abs_builddir)
